lib/manifest-generator.js

```diff
@@ -27,47 +28,23 @@ var ManifestGenerator = Addon.extend({
    * specified manifestOptions.
    */
   postprocessTree: function(type, tree) {
-    if (type === 'all') {
-      var generateAssetManifest = require('./generate-asset-manifest'); // eslint-disable-line global-require
-      return generateAssetManifest(tree, this.manifestOptions);
-    }
-
-    return tree;
-  },
+    if (type !== 'all') { return tree; }
 
-  /**
-   * Replace the manifest placeholder with an escaped version of the manifest.
-   * We do this in both the app's index and test's index.
-   */
-  postBuild: function(result) {
-    var options = this.app && this.app.options && this.app.options.assetLoader;
-    if (options && options.noManifest) {
-      return;
-    }
+    var options = this.app && this.app.options && this.app.options;
+    var assetLoaderOptions = options.assetLoader;
 
-    var manifestFile = path.join(result.directory, 'asset-manifest.json');
-    var manifest;
+    var generateAssetManifest = require('./generate-asset-manifest'); // eslint-disable-line global-require
+    var treeWithManifest = generateAssetManifest(tree, this.manifestOptions);
 
-    try {
-      manifest = fs.readJsonSync(manifestFile);
-    } catch (error) {
-      console.warn('\n\nWarning: Unable to read asset-manifest.json from build with error: ' + error)
-      console.warn('Warning: Proceeding without generated manifest; you will need to manually provide a manifest to the Asset Loader Service to load bundles at runtime. If this was intentional you can turn this message off via the `noManifest` flag.\n\n');
-      manifest = { bundles: {} };
+    if (assetLoaderOptions && assetLoaderOptions.noManifest) {
```

Comment:
We can probably change this to be before we generate the asset manifest (likely worthy of another PR since this is the current behavior).